MatrixSDK/Utils/MXHTTPClient.m

```diff
@@ -348,6 +364,87 @@ - (void)tryRequest:(MXHTTPOperation*)mxHTTPOperation
             failure(error);
         }
     }];
+    
+    // Handle SSL certificates
+    [mxHTTPOperation.operation setWillSendRequestForAuthenticationChallengeBlock:^(NSURLConnection *connection, NSURLAuthenticationChallenge *challenge) {
+        
+        NSURLProtectionSpace *protectionSpace = [challenge protectionSpace];
+        
+        if ([protectionSpace.authenticationMethod isEqualToString:NSURLAuthenticationMethodServerTrust])
+        {
+            if ([httpManager.securityPolicy evaluateServerTrust:protectionSpace.serverTrust forDomain:protectionSpace.host])
+            {
+                NSURLCredential *credential = [NSURLCredential credentialForTrust:protectionSpace.serverTrust];
+                [[challenge sender] useCredential:credential forAuthenticationChallenge:challenge];
+            }
+            else
+            {
+                NSLog(@"[MXHTTPClient] Shall we trust %@?", protectionSpace.host);
+                
+                if (onUnrecognizedCertificateBlock)
+                {
+                    SecTrustRef trust = [protectionSpace serverTrust];
+                    CFIndex index = SecTrustGetCertificateCount(trust);
+                    
+                    for (int i = 0; i < index; i++)
```

Comment:
I'm not sure we want to be looping through the certificates here, as I think this would mean we'd keep popping up dialogs for the other certs in the chain if the user chose not to trust it (although self signed certs will only have one). Android prompts for the leaf certificate, so I'd suggest just doing that (the one at index 0: https://developer.apple.com/library/mac/documentation/Security/Reference/certifkeytrustservices/#//apple_ref/c/func/SecTrustGetCertificateAtIndex).